sda-commons-server-auth/src/test/java/org/sdase/commons/server/auth/key/RsaKeyLoaderSchedulerTest.java

```diff
@@ -0,0 +1,103 @@
+package org.sdase.commons.server.auth.key;
+
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+import java.security.interfaces.RSAPublicKey;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.junit.After;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class RsaKeyLoaderSchedulerTest {
+
+  private final RsaPublicKeyLoader keyLoader = new RsaPublicKeyLoader();
+  private final ScheduledExecutorService executorService =
+      Executors.newSingleThreadScheduledExecutor();
+  private final RsaKeyLoaderScheduler keyLoaderScheduler =
+      RsaKeyLoaderScheduler.create(keyLoader, executorService);
+  private boolean elseCondition = false;
+
+  @Test
+  public void shouldNotBeAuthenticatedWithRevokedKey() {
+    // given
+    final RSAPublicKey validKey = Mockito.mock(RSAPublicKey.class);
+    final String kidForValidKey = "kidForValidKey";
+    final RSAPublicKey keyToBeRevoked = Mockito.mock(RSAPublicKey.class);
+    final String kidForKeyToBeRevoked = "kidForKeyToBeRevoked";
+
+    KeySource keySource =
+        new KeySource() {
+          @Override
+          public List<LoadedPublicKey> loadKeysFromSource() throws KeyLoadFailedException {
+            if (!elseCondition) {
+              return Arrays.asList(
+                  new LoadedPublicKey(kidForValidKey, validKey, this),
+                  new LoadedPublicKey(kidForKeyToBeRevoked, keyToBeRevoked, this));
+            } else {
+              return singletonList(new LoadedPublicKey("kidForValidKey", validKey, this));
+            }
+          }
+        };
+
+    keyLoader.addKeySource(keySource);
+    keyLoaderScheduler.start(0, 2, TimeUnit.SECONDS);
+    await()
+        .untilAsserted(
+            () -> {
+              assertThat(keyLoader.getKey(kidForValidKey)).isSameAs(validKey);
+              assertThat(keyLoader.getKey(kidForKeyToBeRevoked)).isSameAs(keyToBeRevoked);
+            });
+
+    // when -> simulate revoking of key:
+    elseCondition = true;
+
+    // then
+    await()
+        .untilAsserted(
+            () -> {
+              assertThat(keyLoader.getKey(kidForValidKey)).isSameAs(validKey);
+              // revoked key not found anymore
+              assertThat(keyLoader.getKey(kidForKeyToBeRevoked)).isNull();
+            });
+  }
+
+  @Test
+  public void shouldContinueJobInCaseOfError() {
```

Comment:
Is the name correct? I see an assertion for a successful run followed by the assertion to still have valid data after errors.
The other way around would be helpful as well to verify that the scheduler continues after error. Usually the executor stops it's schedule after an exception is thrown in execution.

Reply:
As @dhenneke reported below, we may only need the other way round and assert that the scheduler continues after error.